integrations/azure-devops/azure_devops/misc.py

```diff
@@ -68,6 +69,15 @@ class AzureDevopsSelector(Selector):
     selector: AzureDevopsSelector
 
 
+class AzureDevopsFileResourceConfig(ResourceConfig):
+    class AzureDevopsSelector(Selector):
+        query: str
+        files: dict[str, Any] = {"path": "", "repos": None, "max_depth": None}
+
+    kind: Literal["file"]
+    selector: AzureDevopsSelector
```

Comment:
this should be defined outside of the class, its not a good practice to define a class in a class. 

also the files object type should be a pydantic class as well with type validation